### PR TITLE
Match: support for supplying multiple app identifiers at once

### PR DESCRIPTION
**Pull request by jverdi**
🔑 Allows you to supply multiple app identifiers to match at once and it will only perform (1) clone of your repo, and (1) check for your associated certificate.

As an action:

```
match(type: "appstore", 
      app_identifier: ["tools.fastlane.app", "tools.fastlane.app2", "tools.fastlane.app3"])
```

I made sure to also continue supporting the old syntax, so this doesn't break existing configurations:

```
match(type: "appstore", app_identifier: "tools.fastlane.app")
```

Multiple apps via command-line:

```
match appstore -a tools.fastlane.app,tools.fastlane.app2,tools.fastlane.app3
```

### CODE REVIEW

**Comment by jverdi** [comment]
This addresses performance issues discussed in #4100

I don't see the need for a full matrix-like solution. Generally speaking, CI builds will be building for one environment at a time. The real pain point has been the proliferation of app ids for extensions.

File: match/README.md [hunk 1]
```diff
@@ -174,20 +174,17 @@ match --help
...
 ```
-match appstore -a tools.fastlane.app
-match appstore -a tools.fastlane.app.watchkitapp
+match appstore -a tools.fastlane.app,tools.fastlane.app.watchkitapp
 ```
 
 You can make this even easier using [fastlane](https://github.com/fastlane/fastlane/tree/master/fastlane) by creating a match lane like this:
 
 ```
 lane :match do
```
Comment (KrauseFx) [inline match/README.md]: Ah, the lane shouldn't be called match actually, I submitted a separate PR to change that https://github.com/fastlane/fastlane/pull/6074, make sure to rebase after it got merged

File: match/lib/match/options.rb [hunk 2]
```diff
@@ -31,7 +31,8 @@ def self.available_options
         FastlaneCore::ConfigItem.new(key: :app_identifier,
                                      short_option: "-a",
                                      env_name: "MATCH_APP_IDENTIFIER",
-                                     description: "The bundle identifier of your app",
+                                     description: "The bundle identifier(s) of your app (comma-separated)",
+                                     is_string: false,
```
Comment (KrauseFx) [inline match/lib/match/options.rb]: You should be able to use
```
type: Array
```

Comment (KrauseFx) [inline match/lib/match/options.rb]: Oh, never mind, the user can pass either an array or String 👍 This is ok then

File: match/lib/match/runner.rb [hunk 3]
```diff
@@ -10,25 +10,45 @@ def run(params)
...
+        app_identifiers = params[:app_identifier]
+      else
+        app_identifiers = params[:app_identifier].split(/\s*,\s*/).uniq
+      end
+
       # Verify the App ID (as we don't want 'match' to fail at a later point)
-      spaceship.bundle_identifier_exists(params) if spaceship
+      app_identifiers.each do |app_identifier|
+        params[:app_identifier] = app_identifier
+        spaceship.bundle_identifier_exists(params) if spaceship
```
Comment (KrauseFx) [inline match/lib/match/runner.rb]: We could just skip the whole looping if no spaceship is defined, right? 👍

**Comment by KrauseFx** [comment]
I love this approach: very little change, and very effective. I wonder if there are any other side effects we should think about?
I added some comments before it's ready to be merged 👍

**Comment by jverdi** [comment]
Re-opened with feedback incorporated as: https://github.com/fastlane/fastlane/pull/6080
